### DISCUSSION TITLE
why emulate riscV

### DISCUSSION THREAD

**Discussion by NathanPeake** [fnuecke/oc2#83] [Q&A]
why emulate riscV? 

why not use something like qemu to run any os on the users native hardware without converting instruction sets. or even docker.

if you did convert to qemu or docker you could add cpu tiers

**Comment by atirutw** [comment]
This might be because with QEMU, there'd be a lot more hassle to deal with especially since QEMU is an external software. While QEMU can emulate RISC-V, using a native Java solution(what the devs are doing) might be more convenient for them.

Idk about Java development tho so take my words with a huge grain of salt.

**Reply by CoolCat467** [reply]
Grain of salt big as moon =^._.^=

**Reply by NathanPeake** [reply]
I was more talking about not emulating riscV altogether and just using the system that the user is on, eg x86-64

**Reply by atirutw** [reply]
> the system that the user is on,

How would you deal with users running on other architectures then? These kind of mod needs a consistent architecture and you wouldn't wanna run anything directly on the user's CPU anyway since that'd allow someone to do malicious stuff through Minecraft.

**Reply by NathanPeake** [reply]
> > the system that the user is on,
> 
> How would you deal with users running on other architectures then? These kind of mod needs a consistent architecture and you wouldn't wanna run anything directly on the user's CPU anyway since that'd allow someone to do malicious stuff through Minecraft.

thats where qemu come in, its a vm system thats cross platform and cross architecture.

**Comment by fnuecke** [comment]
Sedna (the RISC-V emulator bit) originally started independent of this. I just wanted to learn about RISC-V and writing emulators is fun. So that's where that came from. With that in hand, it was tempting to try to make a mod around it for fun (I mean, Linux in Minecraft, come on). And with the big advantage over OC1 that it'd just work on any machine (OC1 needed native libs, compiled for all to-be-supported platforms), as pointed out by @atirut-w.
